tensorrt_llm/_torch/models/modeling_utils.py

```diff
@@ -775,60 +779,76 @@ def _load_weights_impl(model: Union[nn.Module, DecoderModelForCausalLM],
+                if any(skip_module in name for skip_module in skip_modules):
+                    return
+
+                # skip load weights if tie word embeddings is enabled and layer is lm_head
+                if model.config.tie_word_embeddings and name.startswith(
+                        "lm_head"):
+                    return
+
+                # Skip loading weights for embedding and lm_head if LoRA is enabled and has custom values
+                if hasattr(model, "model") and hasattr(
+                        model.model, 'has_custom_embed_tokens'
+                ) and model.model.has_custom_embed_tokens and name == "model.embed_tokens":
+                    return
+                if hasattr(model, 'has_custom_lm_head'
+                           ) and model.has_custom_lm_head and name == "lm_head":
+                    return
+
+                names = name.split('.')
+                # WAR: better solution is that llama has its own load_weights function.
+                if names[-1] == 'next_layer_layernorm':
+                    return
+                if names[-1] in params_map:
+                    module_weights = []
+                    for new_name in params_map[names[-1]]:
+                        fw = filter_weights('.'.join(names[:-1] + [new_name]),
+                                            weights)
+                        if new_name in ['k_proj', 'v_proj']:
+                            num_kv_heads_list = [num_kv_heads
+                                                 ] * len(fw) if isinstance(
+                                                     num_kv_heads,
+                                                     int) else num_kv_heads
+                            fw = {
+                                k:
+                                duplicate_kv_weight(
+                                    weight=v[:],
+                                    num_kv_heads=num_kv_heads_list[i],
+                                    tensor_parallel_size=tp_size)
+                                if k in ["weight", "bias"] else v
+                                for i, (k, v) in enumerate(fw.items())
+                            }
+
+                        module_weights.append(fw)
+                    module.load_weights(weights=module_weights)
                 else:
-                    for n, p in module._parameters.items():
-                        if p is not None:
-                            p.data.copy_(module_weights[n][:])
+                    module_weights = filter_weights(name, weights)
+                    if hasattr(module, 'load_weights'):
+                        module.load_weights(weights=[module_weights])
+                    else:
+                        for n, p in module._parameters.items():
+                            if p is not None:
+                                p.data.copy_(module_weights[n][:])
+        except torch.OutOfMemoryError as e:
```

Comment:
Is the 70B OOM caused by not releasing GPU memory in-time? 

If the memory is enough for loading all the weights, why parallel loading can OOM and serially loading does not?